src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/accession/services/JpaBasicSpringDataRepositoryDatabaseServiceTest.java

```diff
@@ -106,21 +112,51 @@ public void saveUniqueElementsAndFindByHashReturnsEachAccession() {
 
     @Test
     public void saveUniqueElementsAndFindByHashThatDoesNotExistReturnsNothing() {
-        service.insert(Arrays.asList(TEST_MODEL_1, TEST_MODEL_2, TEST_MODEL_3));
+        service.save(Arrays.asList(TEST_MODEL_1, TEST_MODEL_2, TEST_MODEL_3));
 
         List<AccessionWrapper<TestModel, String, String>> results = service.findAllByHash(
                 Arrays.asList("h0"));
         assertEquals(0, results.size());
     }
 
+    @Test
+    public void testUnsaveIfExistPreviousWithSameHash() throws AccessionCouldNotBeGeneratedException {
+        TestTransaction.flagForCommit();
+        service.save(Arrays.asList(new AccessionWrapper("a0", "h1", TestModel.of("something1"))));
+        TestTransaction.end();
+
+        SaveResponse<String> accessions = service.save(Arrays.asList(TEST_MODEL_1, TEST_MODEL_2, TEST_MODEL_3));
+        assertEquals(2, accessions.getSavedAccessions().size());
+        assertEquals(1, accessions.getSaveFailedAccessions().size());
+
+        TestTransaction.start();
+        TestTransaction.flagForCommit();
```

Comment:
can u help me understand what is the use of TestTransaction here , does datajpa doesnt serve transactions.

Reply:
Sorry didn't read it, yeah, it provides, the problem is that because it generates a large transaction elements are not written and the code requires that the elements are previously existent on the database so I force it before, so for this part I need to start it again. This is the main reason why I'm starting to hate @DataJpaTest and thinking about skipping it altogether.